### PR TITLE
Add URLPattern Component testing

### PR DESCRIPTION
**Pull request by jelee53**
#### 8c8bc4180463af204120fae9c28f7e2f13372923
<pre>
Add URLPattern Component testing
<a href="https://bugs.webkit.org/show_bug.cgi?id=282779">https://bugs.webkit.org/show_bug.cgi?id=282779</a>
<a href="https://rdar.apple.com/139471135">rdar://139471135</a>

Reviewed by Chris Dumez.

Add separate testing for URLPattern constructor

* LayoutTests/fast/urlpattern/urlpattern-component-expected.txt: Added.
* LayoutTests/fast/urlpattern/urlpattern-component.html: Added.

Canonical link: <a href="https://commits.webkit.org/286336@main">https://commits.webkit.org/286336@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/35f7ff0683ab9cc2ff0acdfa635d76f2de5cbece

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/75605 "Passed style check") | [⏳ 🛠 ios ](https://ews-build.webkit.org/#/builders/iOS-18-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/28456 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/80085 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/26869 "Built successfully") 
| | [⏳ 🛠 ios-sim ](https://ews-build.webkit.org/#/builders/iOS-18-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/2893 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/59290 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/17480 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/78672 "Passed tests") | [⏳ 🧪 ios-wk2 ](https://ews-build.webkit.org/#/builders/iOS-18-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/64930 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/39652 "Passed tests") | 
| | [⏳ 🧪 ios-wk2-wpt ](https://ews-build.webkit.org/#/builders/iOS-18-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/22415 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/25198 "Built successfully") | 
| | [⏳ 🧪 api-ios ](https://ews-build.webkit.org/#/builders/iOS-18-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/22756 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/81564 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/2944 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/1854 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/67531 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/3095 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/64906 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/66831 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/10793 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/8934 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/11681 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/2901 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/2926 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/3861 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/2933 "Built successfully") | | |

### CODE REVIEW

File: LayoutTests/fast/urlpattern/urlpattern-component.html [hunk 1]
```diff
@@ -0,0 +1,2975 @@
+<!DOCTYPE html>
+<html>
+<head>
+<script src="../../resources/js-test-pre.js"></script>
```
Comment (cdumez) [inline LayoutTests/fast/urlpattern/urlpattern-component.html]: Should just include `js-test.js`, not the legacy `js-test-pre.js`

File: LayoutTests/fast/urlpattern/urlpattern-component.html [hunk 2]
```diff
@@ -0,0 +1,2975 @@
...
+      }
+
+
+  });
+}
+
+executeURLPatternConstructor(jsonData);
+
+</script>
+<script src="../../resources/js-test.js"></script>
```
Comment (cdumez) [inline LayoutTests/fast/urlpattern/urlpattern-component.html]: Not needed.

**Review by cdumez — APPROVED** [review]
LGTM with the 2 fixes I suggested.

File: LayoutTests/fast/urlpattern/urlpattern-component-expected.txt [hunk 3]
```diff
@@ -0,0 +1,3610 @@
+CONSOLE MESSAGE: NotSupportedError: Not implemented.
```
Comment (szewai) [inline LayoutTests/fast/urlpattern/urlpattern-component-expected.txt:1]: Do you know which case / function throws this error?

**Pull request merged**